tests/v2/validation/workloads/workload_test.go

```diff
@@ -260,6 +264,64 @@ func (w *WorkloadTestSuite) TestWorkloadUpgrade() {
 	validateDeploymentUpgrade(w, namespace.Name, upgradeDeployment, "6", "ubuntu", 2)
 }
 
+func (w *WorkloadTestSuite) TestWorkloadPodScaleUp() {
+	subSession := w.session.NewSession()
+	defer subSession.Cleanup()
+
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(w.client, w.cluster.ID)
+	require.NoError(w.T(), err)
+
+	scaleUpDeployment, err := deployment.CreateDeployment(w.client, w.cluster.ID, namespace.Name, 1, "", "", false, false)
+	require.NoError(w.T(), err)
+
+	validateDeploymentScale(w, namespace.Name, scaleUpDeployment, imageName, 1)
+
+	replicas := int32(2)
+	scaleUpDeployment.Spec.Replicas = &replicas
+
+	scaleUpDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, scaleUpDeployment)
+	require.NoError(w.T(), err)
+
+	validateDeploymentScale(w, namespace.Name, scaleUpDeployment, imageName, 2)
+
+	replicas = int32(3)
+	scaleUpDeployment.Spec.Replicas = &replicas
+
+	scaleUpDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, scaleUpDeployment)
+	require.NoError(w.T(), err)
+
+	validateDeploymentScale(w, namespace.Name, scaleUpDeployment, imageName, 3)
+}
+
+func (w *WorkloadTestSuite) TestWorkloadPodScaleDown() {
+	subSession := w.session.NewSession()
+	defer subSession.Cleanup()
+
+	_, namespace, err := projectsapi.CreateProjectAndNamespace(w.client, w.cluster.ID)
+	require.NoError(w.T(), err)
+
+	scaleDownDeployment, err := deployment.CreateDeployment(w.client, w.cluster.ID, namespace.Name, 3, "", "", false, false)
+	require.NoError(w.T(), err)
+
+	validateDeploymentScale(w, namespace.Name, scaleDownDeployment, imageName, 3)
+
+	replicas := int32(2)
+	scaleDownDeployment.Spec.Replicas = &replicas
+
+	scaleDownDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, scaleDownDeployment)
+	require.NoError(w.T(), err)
+
+	validateDeploymentScale(w, namespace.Name, scaleDownDeployment, imageName, 2)
+
+	replicas = int32(1)
+	scaleDownDeployment.Spec.Replicas = &replicas
+
+	scaleDownDeployment, err = deployment.UpdateDeployment(w.client, w.cluster.ID, namespace.Name, scaleDownDeployment)
+	require.NoError(w.T(), err)
+
+	validateDeploymentScale(w, namespace.Name, scaleDownDeployment, imageName, 1)
```

Comment:
can we change the expectedCount to replicas instead and can you also update other places where this was hardcoded?